helm_deploy/hmpps-interventions-service/templates/cronjob-team-kpis.yaml

```diff
@@ -0,0 +1,72 @@
+{{- if .Values.env_details.statistician -}}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kpi-scripts
+data:
+{{ (.Files.Glob "reports/*.sh").AsConfig | indent 2 }}
+
+---
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: team-kpis
+spec:
+  schedule: "0 0 * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          restartPolicy: "Never"
+          volumes:
+          - name: data-extractor-reports
+            configMap:
+              name: data-extractor-reports
+              defaultMode: 0755
+          - name: kpi-scripts
+            configMap:
+              name: kpi-scripts
+              defaultMode: 0755
+          containers:
+          - name: data-extractor-reporting
+            image: ministryofjustice/data-engineering-data-extractor:develop
+            args: ["psql -v ON_ERROR_STOP=1 --tuples-only --expanded --file=/reports/team_kpi_report.sql > 'export/team_kpi_report' && /scripts/send_to_slack.sh 'export/team_kpi_report'"]
+            volumeMounts:
+            - name: data-extractor-reports
+              mountPath: /reports/team_kpi_report.sql
+              readOnly: true
+              subPath: team_kpi_report.sql
+            - name: kpi-scripts
+              mountPath: /scripts/send_to_slack.sh
+              readOnly: true
+              subPath: send_to_slack.sh
```

Comment:
I used these patterns only recently. I learned that:

- During runtime, we make a `/scripts/send_to_slack.sh` file available
- through volume mounts &rarr; volumes
- from the `kpi-scripts` ConfigMap, which contains a key `send_to_slack.sh` and the contents of `helm_deploy/hmpps-interventions-service/reports/send_to_slack.sh` as a value

The end result is... make the file in github available in the runtime env. This is how we get both the report in the PR and the script to the env without packaging it up in Docker